extensions/netty/deployment/src/main/java/io/quarkus/netty/deployment/JBossNettyLoggerFactory.java

```diff
@@ -149,29 +221,466 @@ public boolean isErrorEnabled() {
 
         @Override
         public void error(String msg) {
-            log.error(msg);
+            if (isErrorEnabled()) {
+                log.error(msg);
+            }
         }
 
         @Override
         public void error(String format, Object arg) {
-            log.errorf(format, arg);
+            if (isErrorEnabled()) {
+                FormattingTuple ft = MessageFormatter.format(format, arg);
+                error0(ft.getMessage(), ft.getThrowable());
+            }
         }
 
         @Override
         public void error(String format, Object argA, Object argB) {
-            log.errorf(format, argA, argB);
+            if (isErrorEnabled()) {
+                FormattingTuple ft = MessageFormatter.format(format, argA, argB);
+                error0(ft.getMessage(), ft.getThrowable());
+            }
         }
 
         @Override
         public void error(String format, Object... arguments) {
-            log.errorf(format, arguments);
+            if (isErrorEnabled()) {
+                FormattingTuple ft = MessageFormatter.format(format, arguments);
+                error0(ft.getMessage(), ft.getThrowable());
+            }
         }
 
         @Override
         public void error(String msg, Throwable t) {
+            if (isErrorEnabled()) {
+                error0(msg, t);
+            }
+        }
+
+        private void error0(String msg, Throwable t) {
             log.error(msg, t);
         }
 
     }
 
+    // MessageFormatter class copied from Netty because its methods are package private
```

Comment:
😩 Can we also put in an enhancement request upstream to Netty to make these public? Otherwise it defeats the purpose of having the logger be pluggable after all.

Reply:
Yup: https://github.com/netty/netty/issues/12761

Reply:
Once https://github.com/netty/netty/pull/12763 is in and we have new Netty version in Quarkus that makes the methods public, we can delete the copied code

Reply:
https://github.com/quarkusio/quarkus/pull/28126 now removes the copied code